camus-sweeper/src/main/java/com/linkedin/camus/sweeper/mapreduce/CamusMorphlineAvroKeyJob.java

```diff
@@ -68,27 +68,28 @@ public void configureJob(String topic, Job job)
         // finding the newest file from our input. this file will contain the newest version of our avro
         // schema.
         Schema inputSchema = getOverridenTopicSchema(job, topic);
+        String schemaRegistryHost = getConfValue(job, topic, "camus.sweeper.schema.registry.host");
 
         if (inputSchema == null) {
             inputSchema = getNewestInputSchemaFromSource(job, topic);
         }
 
-        // checking if we have a key schema used for deduping. if we don't then we make this a map only
-        // job and set the key schema
-        // to the newest input schema
-        String keySchemaStr = getConfValue(job, topic, "camus.sweeper.avro.key.schema");
-        Schema keySchema;
-        if (keySchemaStr == null || keySchemaStr.isEmpty()) {
-            job.setNumReduceTasks(0);
-            keySchema = inputSchema;
-        } else {
-            keySchema = new Schema.Parser().parse(keySchemaStr);
-        }
+        // checking if we can download a key schema used for deduping. if we don't then we make this a map only
+        // job and set the key schema to the newest input schema
+        try {
+            URI deduplicateSchemaURI = new URI(schemaRegistryHost + "/" + topic + "/latest.deduplication");
```

Comment:
will latest schema (with more fields) be OK to deduplicate older event (with less fields)?